RFC/src/RFC-0241_AtomicSwapXMR.md

```diff
@@ -0,0 +1,238 @@
+* Signature \\(a_{Sra}\\)
+* Partial Script public keys \\(K_Ssa\\) and \\(K_Sra\\)
+* Public Nonce \\(R_{Ssa}\\) and \\(R_{Sra}\\) for her script_signature
+* Partial Monero public key \\(X_a\\)
+* All of the rest of the challenge information in e_s for the script signature s_{Ssa}
+* All of the rest of the challenge information in e_r for the script signature s_{Sra}
+
+Alice constructs for the happy path \\(a_{Ssa}\\) and \\(b_{Ssa}\\) with
+$$
+\begin{aligned}
+a_{Ssa} &= r_{Ssa_a} \\\\
+b_{Ssa} &= r_{Ssa_b} +  e_{s}(k_{Ssa}) \\\\
+e_s &= \hash{ (R_{Ss} + X_b) \cat \alpha_i \cat \input_i \cat (K_{Ssa} + K_{Ssb}) \cat C_i} \\\\
+R_{Ss} &= r_{Ssa_a} \cdot H + r_{Ssa_b} \cdot G + r_{Ssb_a} \cdot H + r_{Ssb_b} \cdot G \\\\
+\end{aligned}
+\tag{1}
+$$
+But Alice keeps \\(a_{Ssa}\\) and \\(b_{Ssa}\\) to herself for now.
+
+Alice also constructs for the refund path  \\(a_{Sra}\\) and \\(b'_{Sra}\\) with
+$$
+\begin{aligned}
+a_{Sra} &= r_{Sra_a} +  e(v_{i}) \\\\
+b'_{Sra} &= r_{Sra_b} +  e(k_{Sra}+k_i) \\\\
+e_r &= \hash{ (R_{Sr} + (X_a)) \cat \alpha_i \cat \input_i \cat (K_{Sra} + K_{Srb}) \cat C_i} \\\\
+R_{Sr} &= r_{Sra_a} \cdot H + r_{Sra_b} \cdot G + r_{Srb_a} \cdot H + r_{Srb_b} \cdot G \\\\
+X_a = x_a \cdot G \\\\
+\end{aligned}
+\tag{2}
+$$
+
+Bob can now verify Alice's adaptor signature with
+$$
+\begin{aligned}
+a_{Sra} \cdot H + b_{Sra} \cdot G = R_{Sra} + (C_i+K_{Sra})*e_r \\\\
+\end{aligned}
+\tag{2}
+$$
+
+
+Bob also constructs for the Happy path  \\(a_{Ssb}\\) and \\(b'_{Ssb}\\) with
+$$
+\begin{aligned}
+a_{Ssb} &= r_{Ssb_a} +  e(v_{i}) \\\\
+b'_{Ssb} &= r_{Ssb_b} +  e(k_{Ssb}+k_i) \\\\
+e_s &= \hash{ (R_{Sr} + (X_b)) \cat \alpha_i \cat \input_i \cat (K_{Ssa} + K_{Ssb}) \cat C_i} \\\\
+R_{Ss} &= r_{Ssa_a} \cdot H + r_{Ssa_b} \cdot G + r_{Ssb_a} \cdot H + r_{Ssb_b} \cdot G \\\\
+X_b = x_b \cdot G \\\\
+\end{aligned}
+\tag{2}
+$$
+
+Bob constructs for the refund path \\(a_{Srb}\\) and \\(b_{Srb}\\) with
+$$
+\begin{aligned}
+a_{Srb} &= r_{Srb_a} \\\\
+b_{Srb} &= r_{Srb_b} +  e_{s}(k_{Srb}) \\\\
+e_r &= \hash{ (R_{Ss} + X_a) \cat \alpha_i \cat \input_i \cat (K_{Sra} + K_{Srb}) \cat C_i} \\\\
+R_{Ss} &= r_{Sra_a} \cdot H + r_{Sra_b} \cdot G + r_{Srb_a} \cdot H + r_{Srb_b} \cdot G \\\\
+\end{aligned}
+\tag{1}
+$$
+
+Alice can now verify Bob's adaptor signature with
+$$
+\begin{aligned}
+a_{Ssb} \cdot H + b_{Ssb} \cdot G = R_{Ssb} + (C_i+K_{Ssb})*e_s \\\\
+\end{aligned}
+\tag{2}
+$$
+
+### XTR payment
+
+If Alice and Bob is happy with all the committed values up to know. Alice will create a Tari UTXO with the script mentioned above. 
+And because Bob already gave her the required signatures \\(a_{Srb}\\), \\(b_{Srb}\\) and \\(R_{Srb}\\) of the his part 
+of the aggregated signatures, she knows all the required information to spend this after the lock expired. 
+
+### XMR Payment
+
+If Bob cab see that Alice has published the Tari UTXO with the correct script, Bob can go ahead and publish the Monero UTXO
```

Comment:
```suggestion
If Bob cab see that Alice has published the Tari UTXO with the correct script and signature, Bob can go ahead and publish the Monero UTXO
```

Reply:
There is no signature to check yet on the UTXO. It's just sitting there mined as an unspent output. 
The zero-knowledge proofs Bob needs to check for that UTXO, should have been checked before publishing the UTXO.

Reply:
Surely to check that the agreed transaction was published you can't just look for the script, but this is a detail anyway so not important. The point is Bob finds it. Thanks for clarifying.